user-profile-card-redesing

aleksandra-b-t opened the pull request:
### What does this pull request aim to achieve? 💯
Here are the changes to the redesign of the user profile, created based on the design on the figma page.
Elements that were supposed to disappear from the profile I did not remove permanently but commented out. In case there will be used in a different place. 
NOTE: The 'Thanks' button is not finished yet, but this is for future features.

_Please be concise and link any related issue(s):_ This PR resolves #2203 

<!--### 🚨Before submitting this pull request🚨:-->

<!--_Please do **NOT** submit this PR if you have not done the following:_-->

- [✓ ] I have checked that no one else is working on similar changes.
- [✓ ] I have read the latest [**README**](README.md) and understand the development workflow.
- [✓ ] The name of this branch is: **`feature/<branch_name>`** for a regular feature, **`hotfix/<branch_name>`** for a hotfix, and **`service/<branch_name>`** for a microservice, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [ ✓] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [✓ ] There are no merge conflicts.
- [✓ ] There are no console warnings and errors.
- [✓ ] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [ ✓] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

aleksandra-b-t committed:
user-profile-card-redesing

clubkemp reviewed (commented):
You made the PR to merge into the staging branch, you will need to change to be merging into feature/profile-redesign. Thats our collective working branch where we can consolidate everything and test before merging into staging

https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request
![image](https://user-images.githubusercontent.com/5578094/109220081-5704b900-776d-11eb-9086-2774bfbf47f6.png)

clubkemp commented on client/src/components/Button/ThankButton.js:
```diff
@@ -0,0 +1,123 @@
+flex-direction: row;
+justify-content: center;
+align-items: center;
+padding: 8px 38px;
```
@linadinh2602 just pointed out on my code we should be using rems instead of px, I agree :) 

I guess the question would be what if the figma designs have px in them...maybe it's ok then??

aditilamkhede replied:
@clubkemp figma is based on px only, you to change from px to rem and tweak a little. but yes please change all px to rem. In very cases where rem might now work or px works better is fine.

aleksandra-b-t replied:
ok!

clubkemp commented on client/src/components/Button/ThankButton.js:
```diff
@@ -0,0 +1,123 @@
+
+
+
+// import { Modal, Button } from "antd";
```
I think we can get rid of the commented out code, if its a relic we don't need it in here, and if we ever DO need to go back and reference it we can just get it from and an older version

aditilamkhede replied:
Yes get rid of commented code and any extra white spaces/lines.

aleksandra-b-t replied:
ok!

clubkemp reviewed (commented):
could you use throw up some screenshots of the product, or if there is functionality we need to see try using screencastify (or similar) to make a little video or GIF?

aditilamkhede commented:
> https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request

good catch @clubkemp

aditilamkhede commented on client/src/components/Button/ThankButton.js:
```diff
@@ -0,0 +1,123 @@
+import styled from "styled-components";
+import { mq, theme } from "constants/theme";
+
+const primaryColor = theme?.colors?.royalBlue;
```
here why condition?

aleksandra-b-t replied:
this is how was used the main blue color in other files, which makes it easier for me to use name instead of #XXXX, is that ok?

aditilamkhede replied:
yes, I meant remove conditional operator.

aditilamkhede commented on client/src/components/Button/ThankButton.js:
```diff
@@ -0,0 +1,123 @@
+const ThankButton = () => {
+    return(
+        <>
+        <ThanksButton>♡Thanks</ThanksButton>
```
is this an SVG from Figma?? We need to get SVG from figma. Check in existing icons inside src/assets/icons.

aleksandra-b-t replied:
This isn’t figma's svg this button isn’t done

aditilamkhede commented on client/src/components/Feed/MessagesModal/MessageModal.js:
```diff
@@ -15,6 +15,7 @@ import { WebSocketContext } from "context/WebsocketContext";
 import { useTranslation } from "react-i18next";
 import TagManager from "react-gtm-module";
 import GTM from "constants/gtm-tags";
+import { MailruShareButton } from "react-share";
```
are we suing this?? if not remove it.

aleksandra-b-t replied:
interesting, I didn’t add id, should I remove it?

aditilamkhede commented on client/src/components/Feed/MessagesModal/MessageModal.js:
```diff
@@ -161,7 +171,7 @@ const MessageModal = ({
             id={gtmId}
           >
-            <img src={activeemail} alt={"message-icon"} />
+            <img src={activeemail} className='envelopeIcon' alt={"message-icon"} />
```
why this change?

aleksandra-b-t replied:
I had difficulty overwriting some general styles for envelope SVG. I needed to add className to specify which SVG I want to change.

aditilamkhede commented on client/src/components/Feed/MessagesModal/MessageModal.js:
```diff
@@ -71,6 +72,15 @@ const MessageModal = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  // const envelopeColor = () => {
```
remove commented code

aleksandra-b-t replied:
I am still working on conditional rendering (white/blue) envelope depending on the device

aditilamkhede commented on client/src/components/Feed/MessagesModal/MessagesContainer.js:
```diff
@@ -3,6 +3,7 @@ import styled from "styled-components";
 import { mq, theme } from "constants/theme";
 
 const primaryColor = theme?.colors?.royalBlue;
```
why conditional?

aditilamkhede commented on client/src/components/Picture/ProfilePic.js:
```diff
@@ -9,12 +9,14 @@ const InitialDiv = styled.div`
   border-radius: 50%;
-  border: ${(props) => (props.hasPhoto ? "none" : "0.2rem solid #425af2;")}
+  border: none;
+  border: ${(props) => (props.hasPhoto ? "none" : "none")}
```
you have a photo or not border is none right? so do we need this line?

aleksandra-b-t replied:
Yes, I will clean it

aditilamkhede commented on client/src/components/Profile/ProfileComponents.js:
```diff
@@ -116,13 +116,28 @@ export const NamePara = styled.p`
+  font-family: Poppins;
+  font-weight: bold;
+  font-style: normal;
+  font-size: 26px;
```
change all px to rem and serach your color # in the project, if it already exists just use the color variable name

aleksandra-b-t replied:
Ok

aditilamkhede commented on client/src/components/Profile/ProfileComponents.js:
```diff
@@ -210,14 +225,17 @@ export const HelpContainer = styled.div`
   background-color: ${colors.white};
   box-shadow: 0px 0.2rem 2rem rgba(0, 0, 0, 0.01);
-  max-width: 80rem;
+  max-width: 1173px;
```
change to rem

aleksandra-b-t replied:
ok

aditilamkhede commented on client/src/pages/Profile.js:
```diff
@@ -384,21 +390,33 @@ const Profile = ({
             {about && <DescriptionDesktop> {about} </DescriptionDesktop>}
             <IconsContainer>
-              <HelpContainer>
+              <MobileLocation>
```
did you confirm with cami regarding location of address on mobile and desktop?

aleksandra-b-t replied:
yes, confirmed

aditilamkhede reviewed (changes requested):
@aleksandra-b-t , I added my comments. Also I see the profile as this now: 
<img width="309" alt="Screen Shot 2021-02-25 at 6 55 42 PM" src="https://user-images.githubusercontent.com/22331887/109249459-9696c980-779c-11eb-8329-37fba14c1545.png">

My concern is: 

1. Thank you button is not visible? 
2. Also, Why do you have to make changes in existing message container?

linadinh2602 commented on client/src/components/Feed/MessagesModal/MessagesContainer.js:
```diff
@@ -247,34 +270,35 @@ export const PrivateMessageContainer = styled.div`
     props.isFromUserCard
       ? `
-    margin-left: 7.4rem;
+    // margin-left: 7.4rem;
```
this is a comment too and cause confusion

aleksandra-b-t replied:
I will clean it!

linadinh2602 reviewed (changes requested):
I agree with Aditi that there are some syntax just need to make sure it clean

aleksandra-b-t committed:
required changes

aditilamkhede commented on client/src/components/Picture/ProfilePic.js:
```diff
@@ -24,11 +25,14 @@ const InitialDiv = styled.div`
     props.resolution ? props.resolution : mq.tablet.narrow.minWidth}) {
     margin: 0;
-    height: ${(props) => (props.hasPhoto ? `12.2rem` : `12rem`)};
+    height: ${(props) => (props.hasPhoto ? `137px` : `12rem`)};
```
rem?

aditilamkhede reviewed (changes requested):
I couldn't check the view as build url is throwing 505. But few minor changes like some places you are still using ox, if that can be changed to rem.

aleksandra-b-t committed:
final profile card redesign changes

I added last changes of message button (for mobile and desktop). For now the thank button has been removed.

aleksandra-b-t committed:
user profile view changes

When no one is logged.

aleksandra-b-t committed:
small changes

user info card in auto height.

clubkemp commented on client/src/components/Feed/MessagesModal/MessagesContainer.js:
```diff
@@ -230,51 +246,72 @@ export const PrivateMessageContainer = styled.div`
-      border: 0.1rem solid ${theme.colors.royalBlue};
+      position: absolute;
+      top: -3rem;
+      height: 6rem;
```
I still think this is a little big. If I use the original figma numbers of 40px height and 152px width the button is smaller than how it appears in your design.  I used an online converted and did some tweaking and if you use root font-size 10 the conversion is more like 4rem for height, and 15.2rem for width

https://nekocalc.com/px-to-rem-converter

clubkemp reviewed (commented):
Noticed when I was logged in I didn't see the edit profile as a button on desktop, only the pencil icon.

aleksandra-b-t committed:
change shadow

SinghRomika commented:
@aleksandra-b-t  The build link given 
 https://featureprofile-redesign-user-profile-car.fightpandemics.xyz.
This seems to be the link for Fight pandemics user profile not for Doing Gud. The PRD in notion shows the design for Doin Gud not Fight Pandemics. Can you share the Figma designs for this?

SinghRomika commented:
**Date: 03/07/21**
@aleksandra-b-t  Following is the verification details of the build link : 
https://featureprofile-redesign-user-profile-car.fightpandemics.xyz. 
**DESKTOP VERSION**
Version:  Chrome 89 Beta, IOS 14
OS: Windows10, MacOS BigSur
Browsers: Chrome, Safari, Edge
Resolutions: 1920 * 1080
**According to my observation following is working as expected** 
- Background image can stay blue like it currently is instead of the ocean background.
- Profile's photo have white border.
- Location is under name.
- 'Own profile' view in mobile and desktop version stays like in old version (include edit icon).

**According to my observation following is not working as expected** 
-  **Thanks button should be hidden under the message icon according to the PRD**
![Thanks button should be hidden under the message icon according to the PRD](https://user-images.githubusercontent.com/78573508/110252006-3a6c3c00-7f38-11eb-8134-6d3707918747.gif)

- The Background image is not as per the designs in Figma, the edge on the left side is not curved as the edge of the background image on right, they should have symmetry and consistency. I don't know if this issues is relevant to PR#2241 or not, so plz feel free to comment on this one if its not?

![Background image should have symmetry ](https://user-images.githubusercontent.com/78573508/110252123-bb2b3800-7f38-11eb-8e17-33df646509f4.gif)

SinghRomika commented:
**Date: 03/07/21**
@aleksandra-b-t Following is the verification details of the build link :
https://featureprofile-redesign-user-profile-car.fightpandemics.xyz.
**MOBILE VERSION:**
Brands: Samsung, Huawei, iPhone,  iPad.
Device: Galaxy(A71 5G, Note 9, Note 20, Note 10) Huawei P30 Pro, iPhone(12 Pro Max, 12 Pro, 8th Gen)
Browser: Chrome, Samsung Internet, Safari, Chromium
OS: Android, IOS 14.4

**According to my observation following is working as expected**

- The location icon can be a blue circle for now like the web design below
- The location is under description.
- Profile photo doesn’t have border.
- 'Own profile' view in mobile and desktop version stays like in old version (include edit icon)
- 'Own profile' view in mobile version doesn’t include 'envelope button'.


**According to my observation following is not working as expected**
- The Location icon is too close to the Location text, according to the Figma design it should not be that close
![location icon](https://user-images.githubusercontent.com/78573508/110253017-f7609780-7f3c-11eb-9b75-86d54127acca.gif)

SinghRomika commented:
DATE: 03/07/21
@aleksandra-b-t  Is the Navigation Bar ready for this feature? 
According to the PRD the 
-Navigation bar in mobile version stays like in old versions for consistency of view. CAN YOU PLZ COMMENT ON THIS?

aditilamkhede commented:
> * **Thanks button should be hidden under the message icon according to the PRD**
@SinghRomika I didn't get this. This is not clear. Button should be hidden and it is not there right? So, what's the issue?

aditilamkhede commented:
> * The Background image is not as per the designs in Figma, the edge on the left side is not curved as the edge of the background image on right, they should have symmetry and consistency. I don't know if this issues is relevant to PR#2241 or not, so plz feel free to comment on this one if its not?

@SinghRomika For this, I asked Cami and she said we can ignore that for now.

aditilamkhede commented:
> DATE: 03/07/21
> @aleksandra-b-t Is the Navigation Bar ready for this feature?
> According to the PRD the
> -Navigation bar in mobile version stays like in old versions for consistency of view. CAN YOU PLZ COMMENT ON THIS?

@SinghRomika Could you please elaborate on this? Which Navigation Bar?

SinghRomika commented:
> > * **Thanks button should be hidden under the message icon according to the PRD**
> >   @SinghRomika I didn't get this. This is not clear. Button should be hidden and it is not there right? So, what's the issue?

Date: 03/08/21
@aditilamkhede Seems like the Thanks Button is not showing up since the THANKS feature is still under development and it is a future feature.

SinghRomika commented:
> > DATE: 03/07/21
> > @aleksandra-b-t Is the Navigation Bar ready for this feature?
> > According to the PRD the
> > -Navigation bar in mobile version stays like in old versions for consistency of view. CAN YOU PLZ COMMENT ON THIS?
> 
> @SinghRomika Could you please elaborate on this? Which Navigation Bar?

Date: 03/08/21
@aditilamkhede I wasn't sure about which navigation bar i am suppose to test? I confirmed with @aleksandra-b-t  and its this one 
![Navigation Bar](https://user-images.githubusercontent.com/78573508/110368176-846b2580-7ffd-11eb-8f54-10e059046e01.gif)

SinghRomika commented:
@aleksandra-b-t Thanks for showing the navigation bar. I have verified the Navigation bar in mobile version, it looks as expected.
 The Navigation bar in mobile version stays like in old versions for consistency of view.

SinghRomika commented:
> > * The Background image is not as per the designs in Figma, the edge on the left side is not curved as the edge of the background image on right, they should have symmetry and consistency. I don't know if this issues is relevant to PR#2241 or not, so plz feel free to comment on this one if its not?
> 
> @SinghRomika For this, I asked Cami and she said we can ignore that for now.

Date:03/08/21
@aditilamkhede  Ok!

aditilamkhede committed:
Merge branch 'feature/profile-redesign' into feature/profile-redesign-user-profile-card

aleksandra-b-t committed:
mobile location fixed

SinghRomika commented:
Thanks!@aleksandra-b-t @aditilamkhede 
 The issue persisting in the **MOBILE VERSION** earlier seems to be working as expected  according to my observations made on multiple Mobile devices:
OS: Android(11.0, 10.0, 9.0 ) IOS: 14.4
Device: Samsung( A71 5G, Note 20, Tab S7 ) Huawei P30 Pro, iPhone 12 Pro Max, iPad Air 4th Gen
Browser: Chrome, Safari.

The Location icon is according to the Figma design.
![Location icon PR#2241](https://user-images.githubusercontent.com/78573508/110691919-9afab380-819a-11eb-96e5-8d21f1542f36.gif)

aditilamkhede commented on client/src/components/Picture/ProfilePic.js:
```diff
@@ -24,11 +25,14 @@ const InitialDiv = styled.div`
+    height: ${(props) => (props.hasPhoto ? `137px` : `12rem`)};
     line-height: ${(props) => (props.hasPhoto ? `10rem` : `11rem`)};
-    width: ${(props) => (props.hasPhoto ? `12.2rem` : `12rem`)};
+    width: ${(props) => (props.hasPhoto ? `137px` : `12rem`)};
```
is it possible to change this to 
rem?

aleksandra-b-t replied:
sure!

aditilamkhede reviewed (approved):
@aleksandra-b-t  LGTM, just some minor changes

AndersonNY commented:
https://fightpandemics.slack.com/files/U01NUUC4MDM/F01R45343R9/image_from_ios.jpg  Tested on iPhone 11 and it seems working as expected.

aleksandra-b-t committed:
changes px to rem

aditilamkhede reviewed (approved):
LGTM

clubkemp reviewed (approved):
Looing good!

AndersonNY commented:
The Location icon is correct according to the Figma design. I checked the mobile version with, Samsun iPhone 12 Pro Max, iPhone XS Max, iPhone 6 plus, LG
<img width="363" alt="Screen Shot 2021-03-15 at 12 03 41 PM" src="https://user-images.githubusercontent.com/56901619/111184069-f821ab00-8586-11eb-9582-bc170cfa4f33.png">
<img width="383" alt="Screen Shot 2021-03-15 at 12 07 38 PM" src="https://user-images.githubusercontent.com/56901619/111184170-12f41f80-8587-11eb-8933-628924bc8442.png">

aleksandra-b-t committed:
Merge branch 'feature/profile-redesign' into feature/profile-redesign-user-profile-card

aleksandra-b-t commented:
feature ready

AndersonNY commented:
<img width="1308" alt="Screen Shot 2021-03-17 at 10 35 02 PM" src="https://user-images.githubusercontent.com/56901619/111565066-28c53880-8771-11eb-9a7d-3cae08ce61cd.png">
<img width="1307" alt="Screen Shot 2021-03-17 at 10 35 14 PM" src="https://user-images.githubusercontent.com/56901619/111565069-295dcf00-8771-11eb-9926-1f191d614226.png">

Request  and offers are working as expected.
